components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth/OAuthServiceTest.java

```diff
@@ -0,0 +1,383 @@
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDAO;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDO;
+import org.wso2.carbon.identity.oauth.dao.OAuthConsumerDAO;
+import org.wso2.carbon.identity.oauth.dto.OAuthConsumerDTO;
+import org.wso2.carbon.user.core.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.powermock.api.mockito.PowerMockito.whenNew;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+/**
+ * Unit tests for OAuthService.
+ */
+@PowerMockIgnore({"javax.net.*", "javax.security.*", "javax.crypto.*"})
+@PrepareForTest({OAuthConsumerDAO.class, OAuthService.class, MessageContext.class, MultitenantUtils.class,
+IdentityTenantUtil.class, OAuthAppDAO.class})
+public class OAuthServiceTest extends PowerMockTestCase {
+
+    private static final String LATEST_TIMESTAMP = "1506888000000"; // 01-12-2017:00.00.00 GMT in milli-seconds
+    private static final String OAUTH_LATEST_TIMESTAMP = "OAUTH_LATEST_TIMESTAMP";
+    private static final String OAUTH_NONCE_STORE = "OAUTH_NONCE_STORE";
+
+    @DataProvider(name = "consumerSecret")
+    public Object[][] consumerSecret() {
+        return new Object[][] {{"consumer-secret"}, {null}};
+    }
+
+    @DataProvider(name = "testIsOAuthConsumerValid")
+    public Object[][] isOAuthConsumerValidFlows() {
+        return new Object[][] {{true, true}, {true, false}, {false, true}, {false, false}};
+    }
+
+    @DataProvider(name = "testAuthorizeOauthRequestTokenException")
+    public Object[][] AuthorizeOauthRequestTokenExceptionFlows() {
+        return new Object[][] {{false, true}, {false, false}};
+    }
+
+    @Mock
+    OAuthConsumerDAO oAuthConsumerDAO;
+
+    @Mock
+    ServiceContext serviceContext;
+
+    @Mock
+    MessageContext messageContext;
+
+    @Mock
+    UserRealm userRealm;
+
+    @Mock
+    UserStoreManager userStoreManager;
+
+    @Mock
+    OAuthAppDAO oAuthAppDAO;
+
+    @Test(dataProvider = "testIsOAuthConsumerValid")
+    public void testIsOAuthConsumerValid(boolean shouldURLEncodeSignature, boolean shouldNonceStoreExist) throws
+            Exception {
+
+        String consumerSecret = "consumer-secret";
+
+        // Create OAuthConsumer.
+        OAuthConsumerDTO oAuthConsumer = new OAuthConsumerDTO();
+        oAuthConsumer.setOauthConsumerKey("consumer-key");
+        oAuthConsumer.setOauthNonce("oauth-nonce");
+        oAuthConsumer.setOauthTimeStamp("1506889800000"); // 01-10-2017:20.30.00 GMT in milli-seconds
```

Comment:
Any specific reason for using a hard coded timestamp ?

Reply:
This is just a request parameter same as "consumer-key", "oauth-nonce" ,and etc.